platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/DefaultTransactionPrehandler.java

```diff
@@ -102,5 +104,9 @@ public void prehandleApplicationTransactions(@NonNull final PlatformEvent event)
 
             preHandleTime.update(startTime, time.nanoTime());
         }
+
+        // TODO adapt this logic to read transactions directly from the callback passed in SwirldState.preHandle() when
```

Comment:
did you check with services if the prehandle call is synchronous?

Reply:
Services confirmed they execute preHandle in parallel, which can be seen [here](https://github.com/hashgraph/hedera-services/blob/93693c3adb38c668ebaeb02f3e86e9b06c89475a/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/prehandle/PreHandleWorkflowImpl.java#L141). Each preHandle call should have a separate synchronous execution and blocks until it fully executes.

I think we should be good.